gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1207,6 +1207,9 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 										"incompatible types in binary expression: %v %v %v",
 										lt.TypeID(), n.Op, rt.TypeID()))
 								}
+								// convert untyped to typed
+								checkOrConvertType(store, last, &n.Left, defaultTypeOf(lt), false)
```

Comment:
One small suggestion: move the Left conversion to immediately after the `if liu {` and then delete the `else` clause.

Reply:
I don’t have a strong opinion on this; although it’s somewhat verbose, I find it clearer.

Reply:
Just a suggestion to put a small dent in removing one of the many unnecessary conditional blocks that exist in our code base.  I won't hold up the PR because of it, so I'll resolve this.